src/components/vm/hostdevs/hostDevAdd.jsx

```diff
@@ -0,0 +1,212 @@
+    devicesWithCorrectType.filter(dev => dev.capability && dev.capability.type === type).forEach(nodeDev => {
+        let deviceIsAlreadyAttached = false;
+
+        vm.hostDevices.forEach(hostDev => {
+            const foundNodeDevice = findHostNodeDevice(hostDev, nodeDevices);
+
+            if (foundNodeDevice && nodeDev.path === foundNodeDevice.path)
+                deviceIsAlreadyAttached = true;
+        });
+
+        if (!deviceIsAlreadyAttached)
+            devicesNotAlreadyAttached.push({ selected: false, nodeDev });
+    });
+
+    return devicesNotAlreadyAttached;
+}
+
+const AddHostDev = ({ idPrefix, vm, nodeDevices, close }) => {
+    const [type, setType] = useState("usb_device");
+    const [selectableDevices, setSelectableDevices] = useState([]);
+    const [dialogError, setDialogError] = useState("");
+    const [dialogErrorDetail, setDialogErrorDetail] = useState("");
+
+    useEffect(() => {
+        setSelectableDevices(getSelectableDevices(nodeDevices, vm, type));
+    }, []);
+
+    const setTypeWrapper = (newType) => {
+        setSelectableDevices(getSelectableDevices(nodeDevices, vm, newType));
+        setType(newType);
+    };
+
+    const attach = () => {
+        Promise.all(selectableDevices.map(device => {
+            console.log(device);
+            if (device.selected)
+                return attachHostDevice({ connectionName: vm.connectionName, vmName: vm.name, live: vm.state !== "shut off", dev: device.nodeDev });
+        }))
+                .then(() => {
+                    getVm({ connectionName: vm.connectionName, id: vm.id });
+                    close();
+                })
+                .catch(exc => {
+                    setDialogError(_("Host device could not be attached"));
+                    setDialogErrorDetail(exc.message);
+                });
+    };
+
+    const body = (
+        <Form isHorizontal>
+            <TypeRow idPrefix={idPrefix} type={type} setType={setTypeWrapper} />
+            <DevRow idPrefix={idPrefix} type={type} selectableDevices={selectableDevices} setSelectableDevices={setSelectableDevices} />
+        </Form>
+    );
+
+    const footer = (
+        <>
+            {dialogError &&
+                <ModalError dialogError={dialogError}
+                            dialogErrorDetail={dialogErrorDetail} />}
+            <Button id={`${idPrefix}-attach`}
+                    variant="primary"
+                    onClick={attach}>
+                {_("Add")}
+            </Button>
+            <Button id={`${idPrefix}-cancel`}
+                    variant="link"
+                    className="btn-cancel"
+                    onClick={close}>
+                {_("Cancel")}
+            </Button>
+        </>
+    );
+
+    return (
+        <Modal position="top"
+               variant="medium"
+               id={`${idPrefix}-dialog`}
+               onClose={close}
+               title={_("Add host device to VM")}
```

Comment:
Please let's be consistent, how we use 'Add disk', we probably don't need 'to VM' here.